fe/src/views/header.view.tsx

```diff
@@ -1,63 +1,110 @@
 import { Header, If } from "../utils/garage";
-import { LanguageOptions, activeLanguage } from "../utils/language";
 import { LanguageToggleView } from "./language-toggle.view";
 import { route } from "preact-router";
 import { Tab, Tabs } from "../utils/fe-utils";
+import { PageProps } from "../utils/page-wrapper";
 import { useState } from "preact/hooks";
 
 const selectClass = (input: string): { class: string; selected: boolean } => {
   const baseClass = `rounded-xl p-1rem`;
-
   if (window.location.pathname.includes(input)) {
     return { class: `bg-#0046FF font-normal ${baseClass}`, selected: true };
   } else {
     return { class: baseClass, selected: false };
   }
 };
-
 const getRoute = (input: string) => {
-  if (input === "世界") return "world";
-  if (input === "众生") return "people";
-  if (input === "我们") return "selves";
-
+  if (input === "world") return "world";
+  if (input === "people") return "article?people=1";
+  if (input === "selves") return "selves";
   return input;
 };
 
-export function HeaderView() {
-  const [_, setPath] = useState(window.location.pathname);
-
+export function HeaderView({ routerInfo }: PageProps) {
+  const [inputResearchValue, setInputResearchValue] = useState<string>("");
+  const [rightSideNavbarValue, setRightSideNavbarValue] =
+    useState<boolean>(false);
+  const languageLabel = routerInfo.lang?.toUpperCase();
   const Options = (props: { en: string; zh: string }) => {
     const { en, zh } = props;
     return (
-      <Tab class={selectClass(en).class} selected={selectClass(en).selected}>
-        <If condition={LanguageOptions[activeLanguage.value]}>
+      <Tab
+        value={en}
+        class={selectClass(en).class}
+        selected={selectClass(en).selected}
+      >
+        <If condition={languageLabel}>
           <div slot="EN">{en.toUpperCase()}</div>
           <div slot="ZH">{zh}</div>
         </If>
       </Tab>
     );
   };
 
+  const handleInputChange = (e: Event) => {
+    const target = e.target as HTMLInputElement;
+    setInputResearchValue(target.value);
+  };
+
+  const handleEnterPress = (e: KeyboardEvent) => {
+    if (e.key === "Enter") {
+      route(
+        `/${routerInfo.lang}/article?search=${encodeURIComponent(
+          inputResearchValue
+        )}`
+      );
+    }
+  };
+
+  const checkHeaderInputVisibility = () => {
+    console.log(routerInfo);
+    if (routerInfo.path === "/:lang/world") {
```

Comment:
Please consider reconstruct the below code. 

Seems like we are only returning false when path is `/:lang/world/:area`, if so make `true` default return.

Also are we supposed to return anything when condition is `(Object.keys(routerInfo.matches as {}).includes("search"))`?

Reply:
If you have already put the logic into specific pages, then why not just remove the unused component?

Reply:
ok

Reply:
done